### PR TITLE
More UI friendly labels

### PR DESCRIPTION
**Pull request by kjeremy**

### CODE REVIEW

File: crates/ra_assists/src/assists/replace_if_let_with_match.rs [hunk 1]
```diff
@@ -42,7 +42,7 @@ pub(crate) fn replace_if_let_with_match(ctx: AssistCtx<impl HirDatabase>) -> Opt
         ast::ElseBranch::IfExpr(_) => return None,
     };
 
-    ctx.add_assist(AssistId("replace_if_let_with_match"), "replace with match", |edit| {
+    ctx.add_assist(AssistId("replace_if_let_with_match"), "Replace with Match", |edit| {
```
Comment (matklad) [inline crates/ra_assists/src/assists/replace_if_let_with_match.rs]: Let's add an assert to `add_assist` method that checks the casing. 

Also, is it correct to captitalize *every* word (as opposed to only the first one)?

Comment (matklad) [reply crates/ra_assists/src/assists/replace_if_let_with_match.rs]: IntelliJ caps only the first word, which is what I'd expect as well:

image

Comment (kjeremy) [reply crates/ra_assists/src/assists/replace_if_let_with_match.rs]: > Let's add an assert to `add_assist` method that checks the casing.

I'm not sure we want that. For instance we can refernce code in assists like: `Move 'x' to outer scope` or `Extract 'bar' to function`.

Comment (matklad) [reply crates/ra_assists/src/assists/replace_if_let_with_match.rs]: But the first letter should *always* be a capital one,now?

Comment (kjeremy) [reply crates/ra_assists/src/assists/replace_if_let_with_match.rs]: Yep I think that's true.

File: crates/ra_assists/src/assist_ctx.rs [hunk 2]
```diff
@@ -84,6 +84,12 @@ impl<'a, DB: HirDatabase> AssistCtx<'a, DB> {
         f: impl FnOnce(&mut AssistBuilder),
     ) -> Option<Assist> {
         let label = AssistLabel { label: label.into(), id };
+        assert_eq!(
+            label.label.chars().nth(0).and_then(|c| Some(c.is_uppercase())),
```
Comment (matklad) [inline crates/ra_assists/src/assist_ctx.rs]: Might as well `unwrap` here, an empty label should be treated as an error I think?

Comment (matklad) [reply crates/ra_assists/src/assist_ctx.rs]: r=me otherwise!

File: crates/ra_assists/src/assist_ctx.rs [hunk 3]
```diff
@@ -84,6 +84,12 @@ impl<'a, DB: HirDatabase> AssistCtx<'a, DB> {
         f: impl FnOnce(&mut AssistBuilder),
     ) -> Option<Assist> {
         let label = AssistLabel { label: label.into(), id };
+        assert_eq!(
+            label.label.chars().nth(0).and_then(|c| Some(c.is_uppercase())).unwrap(),
```
Comment (bjorn3) [inline crates/ra_assists/src/assist_ctx.rs]:
```suggestion
         assert!(label.label.chars().nth(0).unwrap().is_uppercase());
```

Comment (kjeremy) [reply crates/ra_assists/src/assist_ctx.rs]: Oh that's much better. Will do.

Comment (sinkuu) [reply crates/ra_assists/src/assist_ctx.rs]: It can also be `label.label.starts_with(char::is_uppercase)`, though this ignores empty strings.

**Pull request merged**